File: Proximidad de Razas/Proximidad de Razas.pas
program Hd;
uses
SysUtils;

var
oldPos: array of Int64;
N, K, maxId, i, x: Int64;

begin
    maxId := -1;
    readln(N, K);
    SetLength(oldPos, 1000001);
    for i := 1 to N do
    begin
        readln(x);
        if oldPos[x] <> 0 then
        begin
            if (i - oldPos[x]) <= K then
            begin
            if x > maxId then
                maxId := x;
            end;
        end;
        oldPos[x] := i;
    end;
    writeln(maxId);
end.

File: Proximidad de Razas/Proximidad de Razas.cpp
#include<stdio.h>
#include<vector>
#define Hd main()

Hd
{
    std::vector<long long> oldPos(1000001,-1);
    long long N,K,maxId=-1;
    scanf("%lld%lld",&N,&K);
    for(long long i=1;i<=N;i++){
        long long x;
        scanf("%lld",&x);
        if(oldPos[x]!=-1){
            if(i-oldPos[x]<=K){
                if(x>maxId)maxId=x;
            }
        }
        oldPos[x]=i;
    }
    printf("%lld\n",maxId);
}